### ISSUE TITLE
How to generate Non-Broken bengali synthetic data for text recognition?

### ISSUE THREAD

**Issue by mobassir94** [clovaai/synthtiger#25]
thank you for your awesome sharing. i tried to generate text recognition data using your library for bangla printed documents,i made all the necessary changes to make this synthesis engine working for bengali text recognition data generation, however for bangla it is breaking text during text to image conversion just like this trdg issue of mine : https://github.com/Belval/TextRecognitionDataGenerator/issues/253 (where the authors of that repo is not responsive at all)

here is a sample that i got while using this synthesis engine.

label : নির্লজ্জ 

image : https://i.ibb.co/0Y0W5GG/image.jpg ( নির্লজ্জ  word got broken )

do you know how to solve this issue? thanks a lot in advance for your great work.

**Comment by moonbings** [comment]
@mobassir94 
Hi, thanks for reporting the bug and sorry for the late reply.

I think this problem is related to ligature.
SynthTIGER engine basically splits text into graphemes and renders them, but this approach doesn't consider ligature.

```python
import regex

graphemes = regex.findall(r"\X", "নির্লজ্জ")
print(graphemes)

# result: ['নি', 'র্', 'ল', 'জ্', 'জ']
# expected: ['নি', 'র্ল' 'জ্জ']
```

The easiest way to avoid this problem is not to split text, but it cannot render curved text.

https://github.com/clovaai/synthtiger/blob/87f6de877d8497d3f3b14d414bd3b0ea5f017175/examples/synthtiger/template.py#L172
https://github.com/clovaai/synthtiger/blob/87f6de877d8497d3f3b14d414bd3b0ea5f017175/examples/synthtiger/template.py#L196

```python
# chars = utils.split_text(label, reorder=True)
chars = [label]
```

So I searched ligatures used in Indian language and found these data.
This data contains information such as virama, consonant stacker, and number joiner.

https://www.unicode.org/Public/14.0.0/ucd/IndicSyllabicCategory.txt

I don't know using this data is best solution, but maybe it alleviates this problem.

| image | label |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/12423224/167308399-ccd2ab67-c1bd-4b16-9e77-b57ee48ee86c.jpg" height="64px" /> | `নির্লজ্জ` |
| <img src="https://user-images.githubusercontent.com/12423224/167308583-da7d3aa0-e14b-4097-9500-35fbc8f6c5b9.jpg" height="64px" /> | `দৃষ্টিভঙ্গি` |
| <img src="https://user-images.githubusercontent.com/12423224/167308713-d16feda8-91d3-4d07-9532-770bbf6e83ee.jpg" height="32px" /> | `হৃদয়ের চঞ্চলতা বন্ধে ব্রতী হলে` |

The bug was fixed in version 1.0.2.
Please update SynthTIGER to latest version.

Thanks.

**Comment by tanvirbKash** [comment]
![Screenshot 2023-04-12 at 2 12 56 PM](https://user-images.githubusercontent.com/82134395/231395138-fb60b932-9cca-438f-9ac6-3df5e283f92a.png)
I am trying to render the word অ্যাঞ্জেলিনা but it gives broken Bengali output. I have tried with first synthtiger 1.2.1 then 1.0.2
I am using synthtiger through synthdog to create synthetic Bangla dataset. I am using command `synthtiger -o ./outputs/SynthDoG_bn -c 5 -w 4 -v template.py SynthDoG config_bn.yaml`  where the yaml file directs to static resources and configurations.


How do I solve this issue? @moonbings @mobassir94 Please help
